Sources/GRPCReflectionService/Server/ReflectionService.swift

```diff
@@ -313,34 +339,63 @@ internal final class ReflectionServiceProvider: Reflection_ServerReflectionAsync
 extension Reflection_ServerReflectionResponse {
   init(
     request: Reflection_ServerReflectionRequest,
-    fileDescriptorResponse: Reflection_FileDescriptorResponse
+    listServicesResponse: Reflection_ListServiceResponse
   ) {
     self = .with {
       $0.validHost = request.host
       $0.originalRequest = request
-      $0.fileDescriptorResponse = fileDescriptorResponse
+      $0.listServicesResponse = listServicesResponse
     }
   }
 
   init(
     request: Reflection_ServerReflectionRequest,
-    listServicesResponse: Reflection_ListServiceResponse
+    result: Result<[Data], GRPCStatus>
   ) {
-    self = .with {
-      $0.validHost = request.host
-      $0.originalRequest = request
-      $0.listServicesResponse = listServicesResponse
+    switch result {
```

Comment:
We have a bunch of repeated code in all of these `init`s. We can simplify this quite a lot and remove a lot of the redundancy and avoid switching in a bunch of places.

The thing that all responses have in common is the `OneOf_MessageResponse` and the original request, so we can work backwards from that. Our starting point in most cases is `Result<X, GRPCStatus>` (and if not, that's very easy to get to).

The first step we can do is getting from `X` to `Reflection_ServerReflectionResponse.OneOf_MessageResponse`, that's quite easy, we don't need helpers for that, e.g.:

```swift
let name = "Foo"
let result = self.registry.extensionsFieldNumbersOfType(named: name).map { fieldNumbers in
  Reflection_ExtensionNumberResponse.with { 
    $0.fieldNumbers = fieldNumbers
    $0.typeName = name
  }
}
```

The next step is getting from `GRPCStatus` as the `Failure` type of the `Result` to `Reflection_ServerReflectionResponse.OneOf_MessageResponse` as the `Success` type. We can do this with an extension on `Result`:

```swift
extension Result where Success == Reflection_ServerReflectionResponse.OneOf_MessageResponse, Failure == GRPCStatus {
  func recover() -> Result<Reflection_ServerReflectionResponse.OneOf_MessageResponse, Never> {
    self.flatMapError { status in
      let error = Reflection_ErrorResponse.with {
        $0.errorCode = Int32(status.code.rawValue)
        $0.errorMessage = status.message ?? ""
      }

      return .success(.errorResponse(error))
    }
  }
}
```

Now we can quite easily wrangle `Result<X, GRPCStatus>` into `Result< Reflection_ServerReflectionResponse.OneOf_MessageResponse, Never>`. 

The next bit is mapping `Reflection_ServerReflectionResponse.OneOf_MessageResponse` into a `Reflection_ServerReflectionResponse`, another extension helps us out here:

```swift
extension Result where Success == Reflection_ServerReflectionResponse.OneOf_MessageResponse {
  func attachRequest(_ request: Reflection_ServerReflectionRequest) -> Result<Reflection_ServerReflectionResponse, Failure> {
    self.map { message in
      Reflection_ServerReflectionResponse.with {
        $0.validHost = request.host
        $0.originalRequest = request
        $0.messageResponse = message
      }
    }
  }
}
```

We can actually combine the above two functions into one which also unwraps the response result into a response:

```swift
extension Result where Success == Reflection_ServerReflectionResponse.OneOf_MessageResponse, Failure == GRPCStatus {
  func makeResponse(request: Reflection_ServerReflectionRequest) -> Reflection_ServerReflectionResponse {
    let result = self.recover().attachingRequest(request)
    // Safe to '!' as the failure type is 'Never'.
    return try! result.get()
  }
}
```

Going back to the original example, this is all we need:

```swift
let name = "Foo"
let result = self.registry.extensionsFieldNumbersOfType(named: name).map { fieldNumbers in
  Reflection_ExtensionNumberResponse.with { 
    $0.fieldNumbers = fieldNumbers
    $0.typeName = name
  }
}

return result.makeResponse(request: request)
```